airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -165,18 +166,26 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
       StandardSyncOutput standardSyncOutput = null;
       try {
         standardSyncOutput = runChildWorkflow(jobInputs);
-
         workflowState.setFailed(getFailStatus(standardSyncOutput));
 
         if (workflowState.isFailed()) {
-          reportFailure(connectionUpdaterInput, standardSyncOutput);
-          prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          final FailureType failureType =
+              standardSyncOutput.getFailures().isEmpty() ? null : standardSyncOutput.getFailures().get(0).getFailureType();
+          reportFailure(connectionUpdaterInput, standardSyncOutput, failureType == FailureType.CONFIG_ERROR ? "Check Failed" + connectionId : null);
+          if (failureType == FailureType.CONFIG_ERROR) {
+            // In the case that the failure is attributable to config_error, we will not retry again
+            // The first time we fail in this way (from a new connection or a success) we will allow 2 attempts,
+            // but only one with repeated failures
+            log.info("Not retrying due to config_error");
```

Comment:
I don't that it is doing what you are expecting here. I would advise to not do that but instead to update the `reportFailure` method to have a non continue mode. if you look at the implementation of the `reportFailure` method, you'll see that the decision of retrying or not is made in this method.